### PR TITLE
Make makefile more general and fix compile warnings

### PR DESCRIPTION
**Pull request by spagnoloG**
Hello fixed compile warnings. And improved Makefile.

### CODE REVIEW

File: src/progressbar.c [hunk 1]
```diff
@@ -150,12 +153,26 @@ void p_iter_destroy(p_iter* it) {
...
+        }
+    } else {
+        // When simply counting iterations
+        if (it->iterations < it->length) {
+            return it->iterations++;
+        } else {
+            // End of iteration
+            return -1; // Special value indicating end
+        }
+    }
```
Comment (RigglePrime) [inline src/progressbar.c:174]: Noooooo my oneliner

File: test/test.c [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #include <stdio.h>
 #include <stdlib.h>
+#include <unistd.h>
```
Comment (RigglePrime) [inline test/test.c:4]: Windows does not have unistd.h unfortunately

**Review by RigglePrime — CHANGES_REQUESTED** [review]

File: src/progressbar.c (hunk 1 above)
Comment (spagnoloG) [reply src/progressbar.c:174]: this change is sadly needed :(. Otherwise compile warnings won't go away.

File: test/test.c (hunk 2 above)
Comment (spagnoloG) [reply test/test.c:4]: will fix thx.

**Review by RigglePrime — APPROVED** [review]

**Pull request merged**